tools/ci/src/ci.rs

```diff
@@ -22,9 +40,14 @@ impl CI {
     /// This is usually related to differing toolchains and configuration.
     pub fn run(self) {
         let sh = xshell::Shell::new().unwrap();
+        let mut prepared_commands = self.prepare(&sh);
 
-        let prepared_commands = self.prepare(&sh);
-
+        if let Some(build_jobs) = self.build_jobs {
+            prepared_commands = prepared_commands
+                .into_iter()
+                .map(|pc| pc.with_env_var("RUST_BUILD_JOBS", build_jobs.to_string().leak()))
```

Comment:
This line will allocate and leak the same string multiple times, once for each command prepared. To avoid this, this should be changed to:

1. Only leak the string once.
2. Remove the `'static` requirement and use the `'a` lifetime that `PreparedCommand` already has.

I'd prefer 2, but it does require a bit more Rust magic.

Reply:
Additionally, this should probably be converted into a normal `for` loop:

```rust
for pc in prepared_commands.iter_mut() {
    pc.env_vars.push(("RUST_BUILD_JOBS", the_string_variant_of_build_jobs));
}
```

Reply:
setting the `RUST_BUILD_JOBS` was a workaround, the best would be passing `--jobs` to cargo

Reply:
`cargo test --jobs 4 -- --test-threads 4` or something like this